### PR TITLE
Migrate the code to Ballerina Swan Lake Preview1

### PR DESCRIPTION
**Pull request by miyurud**
## Purpose
Migrate the code of this repository to Ballerina Swan Lake Preview1 version.

## Goals
Migrate the code of this repository to Ballerina Swan Lake Preview1 version.

## Approach
N/A

## User stories
N/A

## Release note
N/A

## Documentation
N/A

## Training
N/A

## Certification
N/A

## Marketing
N/A

## Automation tests
N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report?no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Samples
N/A

## Related PRs
N/A

## Migrations (if applicable)
N/A

## Test environment
JDK 1.8, Ubuntu 18.04
 
## Learning
N/A

### CODE REVIEW

**Comment by daneshk** [comment]
Need to update travis.yml as well, with the correct url

File: src/slack/tests/main_test.bal [hunk 1]
```diff
@@ -67,7 +67,7 @@ function testPostTextMessage() {
             test:assertFail(msg = response.toString());
         }
     } else {
-        test:assertFail(msg = <string>response.detail()?.message);
+        test:assertFail(msg = <string>response.detail()["message"]);
```
Comment (daneshk) [inline src/slack/tests/main_test.bal:70]:
```suggestion
        test:assertFail(msg = <string>response.message());
```

File: src/slack/tests/main_test.bal [hunk 2]
```diff
@@ -82,23 +82,23 @@ function deleteMessageAfterTest() {
 function testListConversations() {
     var response = convClient->listConversations();
     if (response is error) {
-        test:assertFail(msg = <string>response.detail()?.message);
+        test:assertFail(msg = <string>response.detail()["message"]);
```
Comment (daneshk) [inline src/slack/tests/main_test.bal:85]:
```suggestion
        test:assertFail(msg = <string>response.message());
```

File: src/slack/tests/main_test.bal [hunk 3]
```diff
@@ -82,23 +82,23 @@ function deleteMessageAfterTest() {
...
     } 
 }
 
 @test:Config {}
 function testGetConversationInfo() {
     var response = convClient->getConversationInfo(channelName1);
     if (response is error) {
-        test:assertFail(msg = <string>response.detail()?.message);
+        test:assertFail(msg = <string>response.detail()["message"]);
```
Comment (daneshk) [inline src/slack/tests/main_test.bal:93]:
```suggestion
        test:assertFail(msg = <string>response.message());
```

File: src/slack/tests/main_test.bal [hunk 4]
```diff
@@ -82,23 +82,23 @@ function deleteMessageAfterTest() {
...
     } 
 }
 
 @test:Config {}
 function testGetUserInfo() {
     var response = userClient->getUserInfo(userName);
     if (response is error) {
-        test:assertFail(msg = <string>response.detail()?.message);
+        test:assertFail(msg = <string>response.detail()["message"]);
```
Comment (daneshk) [inline src/slack/tests/main_test.bal:101]:
```suggestion
        test:assertFail(msg = <string>response.message());
```

File: src/slack/tests/main_test.bal [hunk 5]
```diff
@@ -111,7 +111,7 @@ function testGetUserInfo() {
 function testListFiles() {
     var response = fileClient->listFiles(channelName1);
     if (response is error) {
-        test:assertFail(msg = <string>response.detail()?.message);
+        test:assertFail(msg = <string>response.detail()["message"]);
```
Comment (daneshk) [inline src/slack/tests/main_test.bal:114]:
```suggestion
        test:assertFail(msg = <string>response.message());
```

File: src/slack/tests/main_test.bal [hunk 6]
```diff
@@ -122,7 +122,7 @@ function testListFiles() {
 function testGetFileInfo() {
     var response = fileClient->getFileInfo(fileId);
     if (response is error) {
-        test:assertFail(msg = <string>response.detail()?.message);
+        test:assertFail(msg = <string>response.detail()["message"]);
```
Comment (daneshk) [inline src/slack/tests/main_test.bal:125]:
```suggestion
        test:assertFail(msg = <string>response.message());
```

File: src/slack/tests/main_test.bal [hunk 7]
```diff
@@ -131,7 +131,7 @@ function testGetFileInfo() {
 function uploadFileToTest() {
     var response = fileClient->uploadFile(filePath, channelName1);
     if (response is error) {
-        test:assertFail(msg = <string>response.detail()?.message);
+        test:assertFail(msg = <string>response.detail()["message"]);
```
Comment (daneshk) [inline src/slack/tests/main_test.bal:134]:
```suggestion
        test:assertFail(msg = <string>response.message());
```

File: src/slack/tests/main_test.bal [hunk 8]
```diff
@@ -140,7 +140,7 @@ function uploadFileToTest() {
 function deleteFileAfterTest() {
     var response = fileClient->deleteFile(fileId);
     if (response is error) {
-        test:assertFail(msg = <string>response.detail()?.message);
+        test:assertFail(msg = <string>response.detail()["message"]);
```
Comment (daneshk) [inline src/slack/tests/main_test.bal:143]:
```suggestion
        test:assertFail(msg = <string>response.message());
```

File: src/slack/tests/main_test.bal [hunk 9]
```diff
@@ -201,7 +201,7 @@ function testLeaveConveration() {
 function testJoinConversation() {
     var response = convClient->joinConversation(channelName1);
     if (response is error) {
-        test:assertFail(msg = <string>response.detail()?.message);
+        test:assertFail(msg = <string>response.detail()["message"]);
```
Comment (daneshk) [inline src/slack/tests/main_test.bal:204]:
```suggestion
        test:assertFail(msg = <string>response.message());
```

File: src/slack/tests/main_test.bal (hunk 1 above)
Comment (Bhashinee) [reply src/slack/tests/main_test.bal:70]:
```suggestion
        test:assertFail(msg = response.message());
```
Shall we remove `<string>` part as well?

File: src/slack/tests/main_test.bal (hunk 2 above)
Comment (Bhashinee) [reply src/slack/tests/main_test.bal:85]:
```suggestion
        test:assertFail(msg = response.detail()["message"]);
```
Shall we remove `<string>` casting as well?

**Comment by miyurud** [comment]
> Need to update travis.yml as well, with the correct url

Done

File: src/slack/tests/main_test.bal (hunk 2 above)
Comment (miyurud) [reply src/slack/tests/main_test.bal:85]: Done

File: src/slack/tests/main_test.bal (hunk 3 above)
Comment (miyurud) [reply src/slack/tests/main_test.bal:93]: Done

File: src/slack/tests/main_test.bal (hunk 4 above)
Comment (miyurud) [reply src/slack/tests/main_test.bal:101]: Done

File: src/slack/tests/main_test.bal (hunk 5 above)
Comment (miyurud) [reply src/slack/tests/main_test.bal:114]: Done

File: src/slack/tests/main_test.bal (hunk 6 above)
Comment (miyurud) [reply src/slack/tests/main_test.bal:125]: Done

File: src/slack/tests/main_test.bal (hunk 7 above)
Comment (miyurud) [reply src/slack/tests/main_test.bal:134]: Done

File: src/slack/tests/main_test.bal (hunk 8 above)
Comment (miyurud) [reply src/slack/tests/main_test.bal:143]: Done

File: src/slack/tests/main_test.bal (hunk 9 above)
Comment (miyurud) [reply src/slack/tests/main_test.bal:204]: Done

**Pull request merged**